internal/util/test/controller_manager.go

```diff
@@ -96,8 +96,14 @@ func DeployControllerManagerForCluster(
 
 	// run the controller in the background
 	go func() {
-		defer os.Remove(kubeconfig.Name())
-		fmt.Fprintf(os.Stderr, "INFO: Starting Controller Manager for Cluster %s with Configuration: %+v\n", cluster.Name(), config)
+		defer func() {
+			os.Remove(kubeconfig.Name())
+			if r := recover(); r != nil {
+				logger.Error(fmt.Errorf("%v", r), "Something wrong happened with controller manager")
+				os.Exit(1)
+			}
+		}()
+		logger.Info("Starting Controller Manager", "cluster", cluster.Name(), "config", config)
 		if err := rootcmd.RunWithLogger(ctx, &config, logger); err != nil {
 			panic(err)
```

Comment:
Can't we just use the logger here instead of `panic` & `recover`?

Reply:
I put more thought into it and reworked it. Since it is part of test suit, catching all panics that may occur in `RunWithLogger ` is not a desired behavior, thus calling recover is not needed (it was the initial purpose when I introduced it).  Having different log formats may be beneficial too, because it makes a distinction between what is logged by the actual controller and what is logged by test suite